prefork/prefork.go

```diff
@@ -0,0 +1,179 @@
+package prefork
+
+import (
+	"flag"
+	"net"
+	"os"
+	"os/exec"
+	"runtime"
+
+	"github.com/valyala/fasthttp"
+	"github.com/valyala/fasthttp/reuseport"
+)
+
+const preforkChildFlag = "prefork-child"
+const defaultNetwork = "tcp4"
+
+var child bool
+
+// Prefork implements fasthttp server prefork
+//
+// Preforks master process (with all cores) between several child processes
+// increases performance significantly, because Go doesn't have to share
+// and manage memory between cores
+//
+// WARNING: using prefork prevents the use of any global state!
+// Things like in-memory caches won't work.
+type Prefork struct {
+	Addr string
+
+	// The network must be "tcp", "tcp4" or "tcp6".
+	//
+	// By default is "tcp4"
+	Network string
+
+	// Flag to use a listener with reuseport, if not a File Listener will be used
+	//
+	// It's disabled by default
+	Reuseport bool
+
+	ServeFunc         func(ln net.Listener) error
+	ServeTLSFunc      func(ln net.Listener, certFile, keyFile string) error
+	ServeTLSEmbedFunc func(ln net.Listener, certData, keyData []byte) error
+
+	ln    net.Listener
+	files []*os.File
+}
+
+func init() { // nolint:gochecknoinits
+	flag.BoolVar(&child, preforkChildFlag, false, "is child proc")
```

Comment:
This is never going to work be cause [`flag.Parse()`](https://golang.org/pkg/flag/#Parse) is never called.

Reply:
I changed it to `flag.Bool(preforkChildFlag[1:], false, "is a child process")`, because if the user define some extra flags in their program and pass `-prefork-child`, the program crash.

Reply:
The definition of the flag is mandatory